cbr/cbr.go

```diff
@@ -0,0 +1,289 @@
+// Package cbr ...
+package cbr
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/contextbasedrestrictionsv1"
+)
+
+const (
+	kubernetes_service = "containers-kubernetes"
+	is_service         = "is"
+	cos_service        = "cloud-object-storage"
+	keyProtect_service = "kms"
+)
+
+type CBR struct {
+	VPC        []string `json:"VPC,omitempty"`        //The VPCs allowed in the zone. Input in the form 'value,value,...' where value is VPC CRN
+	Address    []string `json:"Address,omitempty"`    //The list of addresses in the zone. Only addresses of type 'ipAddress', 'ipRange', and 'subnet' are allowed in a comma delimited format.
+	ServiceRef []string `json:"ServiceRef,omitempty"` //The service refs in the zone. Input in the form 'value,value,...'
+}
+
+// CBRInterface ...
+type CBRInterface interface {
+
+	// CreateCBRZone ...
+	CreateCBRZone(name string, cbrInput CBR) (string, error)
+
+	// CreateCBRRuleForContainerK8sService ...
+	CreateCBRRuleForContainerK8sService(zoneID string) (string, error)
+
+	// CreateCBRRuleForISService ...
+	CreateCBRRuleForISService(zoneID string) (string, error)
+
+	// CreateCBRRuleForKMSService ...
+	CreateCBRRuleForKMSService(zoneID string) (string, error)
+
+	// CreateCBRRuleForCOSService ...
+	CreateCBRRuleForCOSService(zoneID string) (string, error)
+
+	// CreateCBRRule ...
+	CreateCBRRule(zoneID string, serviceName string) (string, error)
+
+	// DeleteCBRRuleZone ...
+	DeleteCBRRuleZone(ruleID string, zoneID string) error
+}
+
+// CBRInit ...
+type CBRInit struct {
+	accountID                       string
+	resourceGroupID                 string
+	contextBasedRestrictionsService *contextbasedrestrictionsv1.ContextBasedRestrictionsV1
+}
+
+// NewCBRInterface ...
+func NewCBRInterface(apiKey string, accountID string, resourceGroupID string) CBRInterface {
```

Comment:
Why type interface `CBRInterface` being returned here?
This restricts the extension and also following code throw error
```var cbrops   *CBRInit = cbr.NewCBRInterface("xxx", "xxx", "xxx")```

Reply:
@nkkashyap good catch , I missed that also changed the naming convention to make it more meaningfull